src/test/java/org/broadinstitute/hellbender/tools/spark/sv/RunSGAViaProcessBuilderOnSparkUnitTest.java

```diff
@@ -0,0 +1,312 @@
+                sequencesAreCloseEnough = (minDist <= editDistanceTolerance);
+                if(0!=minDist){
+                    System.err.println("Contig that has nonzero edit distance is of length " + Integer.toString(actualString.length()) +
+                                        "\nEdit distance:" + String.valueOf(minDist) +
+                                        "\nactual sequence: " + actualString +
+                                        "\nreverse complement of actual sequence: " + rcOfActualString +
+                                        "\nexpected sequence: " + expectedString);
+                }
+            }
+
+            Assert.assertTrue(sequencesAreTheSame || sequencesAreCloseEnough,
+                              "Culprit contigs (expected, actual) of length: " + String.valueOf(expectedString.length()) + ", " + String.valueOf(actualString.length()) +
+                              "With edit distance " + String.valueOf(minDist));
+        }
+    }
+
+    private static String compareNamesAndComputeSeqEditDist(final File actualFile, final File expectedFile, final boolean fastqFilesWellFormed, final ArrayList<Integer> distances) throws IOException{
+
+        final List<String> actualNames = new ArrayList<>();
+        final List<String> actualSeq = new ArrayList<>();
+        extractNamesAndSeqFromFASTQ(actualFile, fastqFilesWellFormed, actualNames, actualSeq);
+
+        final List<String> expectedNames = new ArrayList<>();
+        final List<String> expectedSeq = new ArrayList<>();
+        extractNamesAndSeqFromFASTQ(expectedFile, fastqFilesWellFormed, expectedNames, expectedSeq);
+
+        Assert.assertEquals(expectedNames, actualNames);
+
+        // in stead of more stringent tests, e.g. "Assert.assertEquals(expectedSeq, actualSeq);",
+        // compute distance and let caller decide how large an edit distance is allowed
+        for(int i=0; i<expectedSeq.size(); ++i){
+            distances.add(StringUtils.getLevenshteinDistance(expectedSeq.get(i), actualSeq.get(i)));
+        }
+
+        //return RunSGAViaProcessBuilderOnSpark.extractBaseNameWithoutExtension(actualFile) + ".fa";
+        return FilenameUtils.getBaseName(actualFile.getName()) + ".fa";
+    }
+
+    // utility function: for extracting read names and sequences from a fastq/fasta file
+    private static void extractNamesAndSeqFromFASTQ(final File FASTAFile, final boolean fastqFilesWellFormed, List<String> readNames, List<String> sequences) throws IOException{
+
+        if(fastqFilesWellFormed){
+            try(final FastqReader reader = new FastqReader(FASTAFile)){
+                while(reader.hasNext()){
+                    final FastqRecord record = reader.next();
+                    readNames.add(record.getReadHeader());
+                    sequences.add(record.getReadString());
+                }
+            }
+        }else{
+            try(final BufferedReader reader = new BufferedReader(new FileReader(FASTAFile))){
```

Comment:
\[FindBugs\] WARNING: Dm: Found reliance on default encoding in org\.broadinstitute\.hellbender\.tools\.spark\.sv\.RunSGAViaProcessBuilderOnSparkUnitTest\.extractNamesAndSeqFromFASTQ\(File, boolean, List, List\): new java\.io\.FileReader\(File\)